### PR TITLE
Revert "Redefine scheduler to set learning rate using recursive formula (#14010)"

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19132 Revert "Redefine scheduler to set learning rate using recursive formula (#14010)"**

This reverts commit 096ee8467cffda68f0ed12a1266cb0010081b809.

See #17913

### CODE REVIEW

**Review by chandlerzuo — APPROVED** [review]

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -257,12 +246,9 @@ def __init__(self, optimizer, T_max, eta_min=0, last_epoch=-1):
         super(CosineAnnealingLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
-        if self.last_epoch == 0:
-            return self.base_lrs
-        return [(1 + math.cos(math.pi * self.last_epoch / self.T_max)) /
-                (1 + math.cos(math.pi * (self.last_epoch - 1) / self.T_max)) *
```
Comment (chandlerzuo) [inline torch/optim/lr_scheduler.py]: I did not catch an edge issue here. The induction here only works when the denominator is none-zero. When it is 0, we should use the legacy formula. Please let me know if I should send a fix diff.

Comment (ezyang) [reply torch/optim/lr_scheduler.py]: If you have time to, that would be great! Otherwise I will see it through.

Comment (chandlerzuo) [reply torch/optim/lr_scheduler.py]: No problem, let me do that. Do you want to land this first, and I send a big diff with the fix? Or abandon this and I send a patch?

Comment (ezyang) [reply torch/optim/lr_scheduler.py]: I'll go ahead and wait for you (abandoning when we land your fix). If you end up not having time I'll land this revert before our next release :)

Comment (chandlerzuo) [reply torch/optim/lr_scheduler.py]: See https://github.com/pytorch/pytorch/pull/19180

**Comment by ezyang** [comment]
Obsoleted by #19180
